### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by milldr**
## what
- Initial implementation of managed grafana module

## why
- This module is responsible for provisioning an Amazon Managed Grafana workspace.

## references
- customer engagement

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -83,40 +67,63 @@ usage: |-
...
+  resource "aws_grafana_role_association" "sso" {
+    for_each = local.enabled ? {
+      for association in var.sso_role_associations : association.role => association
+    } : {}
+
+    role      = each.value.role
+    group_ids = each.value.group_ids
+
+    workspace_id = module.managed_grafana.workspace_id
   }
```
Comment (osterman) [inline README.yaml]: Why not support this in the module directly? It can be optional.

File: main.tf [hunk 2]
```diff
@@ -1,15 +1,91 @@
...
+  count = local.enabled ? 1 : 0
+
+  name        = module.this.id
+  description = "Amazon Managed Grafana for ${module.this.id}"
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  account_access_type      = "CURRENT_ACCOUNT"
```
Comment (osterman) [inline main.tf:14]: This shouldn't be a parameter?

File: README.yaml (hunk 1 above)
Comment (milldr) [reply README.yaml]: okay I'll add it

File: main.tf [hunk 3]
```diff
@@ -1,15 +1,91 @@
...
+      },
+    ]
+  })
+
+  # Allow this role to assume other roles.
+  # Used for cross-account access
+  dynamic "inline_policy" {
+    for_each = length(var.additional_allowed_roles) > 0 ? [1] : []
+    content {
+      name = "${module.this.id}-allowed-roles"
```
Comment (osterman) [inline main.tf]: Not using null label properly. 
```suggestion
      name = "${module.this.id}-allowed-roles"
```

File: main.tf (hunk 2 above)
Comment (milldr) [reply main.tf:14]: the alternate access type, `ORGANIZATION`, requires additional configuration that I haven't tested. It's probably simple, but it should be a feature request

File: main.tf (hunk 3 above)
Comment (milldr) [reply main.tf]: replaced with separate null-label modules for creating sub-labels

File: README.yaml (hunk 1 above)
Comment (milldr) [reply README.yaml]: added

**Review by osterman — APPROVED** [review]

**Pull request merged**